### PR TITLE
Softdevice_controller: remove RNG from the SoftDevice Controller

### PR DESCRIPTION
**Pull request by bewi-nordic**
The application is responsible for providing the SoftDevice Controller
with random number generating functions.

Signed-off-by: Bernhard Wimmer <bernhard.wimmer@nordicsemi.no>

### CODE REVIEW

**Comment by rugeGerritsen** [comment]
@joerchan , should we add the label "CI-all" to this PR?

**Review by rugeGerritsen — APPROVED** [review]

File: subsys/bluetooth/controller/hci_driver.c [hunk 1]
```diff
@@ -447,6 +473,24 @@ static int hci_driver_open(void)
 		return -ENOMEM;
 	}
 
+	entropy_source = device_get_binding(DT_LABEL(DT_NODELABEL(rng)));
+	if (!entropy_source) {
+		BT_ERR("An entropy source is required");
+		return -EINVAL;
```
Comment (joerchan) [inline subsys/bluetooth/controller/hci_driver.c:479]:
```suggestion
		return -ENODEV;
```

File: subsys/bluetooth/controller/hci_driver.c [hunk 2]
```diff
@@ -447,6 +473,24 @@ static int hci_driver_open(void)
...
+
+	sdc_rand_source_t rand_functions = {
+		.rand_prio_low_get = rand_prio_low_vector_get,
+		.rand_prio_high_get = rand_prio_high_vector_get,
+		.rand_poll = rand_prio_low_vector_get_blocking
+	};
+
+	err = sdc_rand_source_register(&rand_functions);
+	if (err) {
+		BT_ERR("Failed to register randomness source with the SoftDevice Controller");
```
Comment (joerchan) [inline subsys/bluetooth/controller/hci_driver.c:490]: Maybe add the err code returned to the log
```suggestion
		BT_ERR("Failed to register rand source (%d)", err);
```

**Review by joerchan — CHANGES_REQUESTED** [review]
Great to finally see this change :)

**Review by joerchan — APPROVED** [review]

**Comment by joerchan** [comment]
> @joerchan , should we add the label "CI-all" to this PR?

Let's be on the safe side.

**Pull request merged**